File: apps/api/gaia-workflow/gaia-workflow-app/src/test/resources/code-examples/java/ExampleCode.java
import java.util.HashMap;
import java.util.Map;

/**
 * Java 代码节点示例。
 * <p>
 * 约束：
 * <ul>
 *   <li>需提供 public 无参构造（默认即满足）</li>
 *   <li>需提供 {@code execute(Map<String,Object>)} 方法，返回 Map 作为节点输出</li>
 *   <li>无需实现 CodeExecute 接口，DynamicCompileNode 会通过反射调用 execute</li>
 * </ul>
 * 输入参数 name（String），输出 greeting / length。
 */
public class ExampleCode {

    public Map<String, Object> execute(Map<String, Object> inputs) {
        Map<String, Object> result = new HashMap<>();
        Object name = inputs.get("name");
        String nameStr = name == null ? "" : name.toString();
        result.put("greeting", "Hello, " + nameStr + "!");
        result.put("length", nameStr.length());
        return result;
    }
}

File: apps/api/gaia-workflow/gaia-workflow-app/src/test/resources/code-examples/javascript/ExampleCode.js
/**
 * JavaScript 代码节点示例（GraalJS 引擎）。
 * <p>
 * 约束：
 * <ul>
 *   <li>代码会被包裹为 function jsFunc(){ &lt;code&gt; }; jsFunc();</li>
 *   <li>最后一个表达式作为返回值，需为对象（会被强转为 Map）</li>
 *   <li>输入参数以顶层变量形式注入 bindings，直接用变量名访问</li>
 * </ul>
 * 输入参数 name（String），输出 greeting / length。
 */
var nameStr = name == null ? "" : name.toString();
var result = {
    greeting: "Hello, " + nameStr + "!",
    length: nameStr.length
};
result;
